src/timesync/timesyncd-manager.c

```diff
@@ -831,21 +831,23 @@ int manager_connect(Manager *m) {
                         bool restart = true;
 
                         /* Our current server name list is exhausted,
-                         * let's find the next one to iterate. First
-                         * we try the system list, then the link list.
-                         * After having processed the link list we
-                         * jump back to the system list. However, if
-                         * both lists are empty, we change to the
-                         * fallback list. */
+                         * let's find the next one to iterate. First we try the system list, then the link list,
+                         * then the runtime list. After having processed the link list we jump back to the
+                         * system list, then to the runtime list. However, if all lists are empty, we change to
+                         * the fallback list. */
                         if (!m->current_server_name || m->current_server_name->type == SERVER_LINK) {
                                 f = m->system_servers;
                                 if (!f)
                                         f = m->link_servers;
+                                if (!f)
```

Comment:
I'm not sure this is right, `f = m->link_servers;` will never be possible because it will simply overwrite with `runtime_servers`.

Reply:
you are not missing anything, apologies.